src/main.rs

```diff
@@ -194,10 +200,13 @@ where
     active_threads.push(
         thread::Builder::new()
             .name("sniffing_handler".to_string())
-            .spawn(move || {
-                while running.load(Ordering::Acquire) {
+            .spawn(move || 'sniffing: loop {
+                for sniffer in sniffers.iter_mut() {
                     if let Some(segment) = sniffer.next() {
-                        network_utilization.lock().unwrap().update(&segment)
+                        network_utilization.lock().unwrap().update(segment);
+                    }
+                    if !running.load(Ordering::Acquire) {
+                        break 'sniffing;
```

Comment:
Hey, I'm sorry but I missed this the first time. I haven't seen this pattern, and actually don't fully understand this syntax. I couldn't find any info about it - could you point me in the right direction? (or maybe add a comment to explain what's going on here?)

Reply:
You're right I couldn't find a reference in the book. Here is some explanation in [Rust by example](https://doc.rust-lang.org/stable/rust-by-example/flow_control/loop/nested.html). Here it's used to break from the outer `loop`. If I were not using the tag, `break` would just exit the `for`, but the loop would repeat again.

Reply:
uff - this is eerily similar syntax to lifetimes. What'chya gonna do though :) Thanks for the link